app/models/event.rb

```diff
@@ -279,4 +279,49 @@ def event_checkout_summary
     )
   end
 
+  # Returns the most checked-out games for an event
+  #
+  # @note
+  #   This doesn't return fully-hydrated game objects. You'll need
+  #   to add extra data to the query inside of here if you need
+  #   all the data typically accessible on a game.
+  #
+  # @example Get the top games for the current event
+  #   Event.current.top_games(1) #=> [
+  #     Game<# id: 73, name: "Monopoly", times_checked_out: 20 >
+  #   ]
+  #
+  # @param limit [Integer] how many top games to return. Defaults to 10.
+  #
+  # @returns [Array<Game>] array of games with id, name, and times_checked_out
+  def top_games(limit=10)
+    placeholder = SecureRandom.uuid.downcase.gsub('-', '')
+    games = Event.connection.execute(
+      <<-SQL
+        select * from (
+          select
+            t.id as title_id
+            ,count(distinct c.id) as checkouts
+          from games g
+          left join (select * from checkouts where event_id = #{self.id}) c on g.id = c.game_id
+          inner join titles t on t.id = g.title_id
+          where
+            g.culled = false
+              or (g.culled = true and g.updated_at::date between '#{self.start_date - 2.days}' and '#{self.end_date + 2.days}')
+          group by 1
+          order by 2 desc, 1
+        ) c
+        where checkouts > 0
+        limit '#{limit}'
+      SQL
+    )
+    games_with_availability = []
+    for game in games
+      game[:available] = Game.copy_available(game["title_id"])
+      game[:title] = Title.find(game["title_id"]).title
```

Comment:
Yikes, this is an `n+1` query. You should batch these and execute them as one "find" and only select the title field. Otherwise you're making a large query and lots of them.